crates/edr_provider/src/data.rs

```diff
@@ -1420,26 +1557,47 @@ mod tests {
                 chain_id: 1,
             });
 
-            let sender = *self
+            TransactionRequestAndSender {
+                request,
+                sender: self.first_local_account(),
+            }
+        }
+
+        fn first_local_account(&self) -> Address {
+            *self
                 .provider_data
                 .local_accounts
                 .keys()
                 .next()
-                .expect("there are local accounts");
+                .expect("there are local accounts")
+        }
+
+        fn impersonated_dummy_transaction(&self) -> anyhow::Result<PendingTransaction> {
+            let mut transaction = self.dummy_transaction_request(None);
+            transaction.sender = self.impersonated_account;
 
-            TransactionRequestAndSender { request, sender }
+            Ok(self.provider_data.sign_transaction_request(transaction)?)
         }
 
         fn signed_dummy_transaction(&self) -> anyhow::Result<PendingTransaction> {
             let transaction = self.dummy_transaction_request(None);
             Ok(self.provider_data.sign_transaction_request(transaction)?)
         }
+    }
 
-        fn impersonated_dummy_transaction(&self) -> anyhow::Result<PendingTransaction> {
-            let mut transaction = self.dummy_transaction_request(None);
-            transaction.sender = self.impersonated_account;
+    struct ConsoleLogTransaction {
```

Comment:
This applies to all related code, but could you please move these to separate files outside of data.rs?

Reply:
Done in [ca9a178](https://github.com/NomicFoundation/hardhat/pull/4659/commits/ca9a178e54f2997856fc25f38c9c43e0f11eaab5)